spec/features/openid_connect/openid_connect_spec.rb

```diff
@@ -534,6 +534,19 @@
     end
   end
 
+  context 'visiting IdP via AAL3 SP' do
```

Comment:
Would it make sense to break these out into their own spec file? This one is already pretty huge. I'm thinking we'll want to add some more tests for some of these weird edge and corner cases so this block could get crowded in an already crowded spec file.